lib/helpers.js

```diff
@@ -26,14 +26,21 @@ function formatTags(tags, telegraf) {
 function overrideTags (parent, child, telegraf) {
   var childCopy = {};
   var toAppend = [];
+  var duplicateKeys = [];
+
   formatTags(child, telegraf).forEach(function (tag) {
     var idx = typeof tag === 'string' ? tag.indexOf(':') : -1;
     if (idx < 1) { // Not found or first character
       toAppend.push(tag);
     } else {
-      childCopy[tag.substring(0, idx)] = tag.substring(idx + 1);
+      if(childCopy[tag.substring(0, idx)]) {
```

Comment:
The whole code should be more generic, for example by making `childCopy[]` an array instead of a string.

Then the rest of the code would then be the same and require no special treatment, regardless if there's one or more value for the given tag, and you wouldn't have to do conditional string manipulation with a hardcoded separator. Would make the whole code shorter and easier to read and review, as well as test.

Reply:
I thought about doing that way, but I didn't want to change much of the original text so I decided to use string, but I agree that with array it's easier to read.

Will work on that.